### PR TITLE
New Voice Channel Drawer

### PR DESCRIPTION
**Pull request by rauchp**
Need to get the final headphones (mute everyone and yourself) functionality going but this is ready for review besides that small ting. 

Tested on iPhone 11 and SE plus device, + a little bit on android sim, but android sim has been unhappy with RoomView (something to do with a flatlist) since 2 days ago? I need to debug that.

I'm going to wait for a final round of nitting with Alex before I merge.

What it looks on iPhone 11 Sim:
Screen Shot 2019-12-07 at 12 54 24 AM

video of muting tested w/ iPhone 11 and iPhone SE sims: https://drive.google.com/file/d/1KqE81r04lRikM1YSOJk5L3Lo4l1e0SxV/view?usp=sharing

### CODE REVIEW

File: civilapp/src/current/VoiceChannelModal.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            <FooterButton image={Deaf} />
+        </View>
+    )
+
+    goToProfile = (id) => {
+        this.props.toggleModalCallback(() => this.props.navigation.navigate('ProfilePrimary', { userId: id }))
+
+    }
+    renderSpeaker = (speaker) => {
+        console.log('speaker:', speaker)
```
Comment (rauchp) [inline civilapp/src/current/VoiceChannelModal.js]: need to remove

File: civilapp/src/current/VoiceChannelModal.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            isMuted = this.state.isMuted
+
+        if (isMuted)
+            isSpeaking = false
+
+        return (
+            <View opacity={(isMuted) ? 0.7 : 1.0} style={styles.speakerView}>
+                <TouchableOpacity onPress={() => this.goToProfile(speaker.id)} style={styles.profileButton}>
+                    <Image source={getAvatarObject(speaker.profileImageUrl).link_chat} style={[styles.profileImage, { borderWidth: isSpeaking ? 3.25 : 0, borderColor: color }]} />
+                    <Text style={{ fontFamily: isSpeaking ? 'SFCompactDisplay-Bold' : 'SFCompactDisplay-Regular', color: 'white', fontSize: isSpeaking ? 19 : 18, marginRight: '2%' }}>{speaker.username}</Text>
```
Comment (carteryu) [inline civilapp/src/current/VoiceChannelModal.js]: bruh moment

File: civilapp/src/current/VoiceChannelModal.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            <FooterButton image={Deaf} />
+        </View>
+    )
+
+    goToProfile = (id) => {
+        this.props.toggleModalCallback(() => this.props.navigation.navigate('ProfilePrimary', { userId: id }))
+
+    }
+    renderSpeaker = (speaker) => {
+        let isMuted = (this.state.selfMutedDict[speaker.id] | this.state.othersMutedDict[speaker.id]) ? true : false
```
Comment (EphemeralX) [inline civilapp/src/current/VoiceChannelModal.js]: is this supposed to be a ||

**Comment by rauchp** [comment]
I just prettier'd my shit up

**Pull request merged**